health-checker/src/test/java/io/stargate/health/CheckerResourceTest.java

```diff
@@ -0,0 +1,179 @@
+  }
+
+  @Test
+  public void readinessAllRegisteredChecks() {
+    Mockito.doReturn(true).when(bundleService).checkBundleStates();
+    Mockito.doReturn(true).when(bundleService).checkDataStoreAvailable();
+
+    healthCheckRegistry.register("test-unhealthy", UNHEALTHY_CHECK);
+    healthCheckRegistry.register("test-ok", OK_CHECK);
+
+    assertThat(checker.checkReadiness(null).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.emptySet()).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(ImmutableSet.of("test-ok", "test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-ok")).getStatus())
+        .isEqualTo(OK.getStatusCode());
+  }
+
+  @Test
+  public void readinessRequiredRegisteredChecks() {
+    Mockito.doReturn(true).when(bundleService).checkBundleStates();
+    Mockito.doReturn(true).when(bundleService).checkDataStoreAvailable();
+
+    healthCheckRegistry.register("test-unhealthy", UNHEALTHY_CHECK);
+    healthCheckRegistry.register("test-ok", OK_CHECK);
+
+    Mockito.when(bundleService.defaultHeathCheckNames())
+        .thenReturn(ImmutableSet.of("test-unhealthy", "test-ok"));
+
+    assertThat(checker.checkReadiness(null).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.emptySet()).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(ImmutableSet.of("test-ok", "test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-ok")).getStatus())
+        .isEqualTo(OK.getStatusCode());
+
+    Mockito.when(bundleService.defaultHeathCheckNames()).thenReturn(ImmutableSet.of("test-ok"));
+
+    assertThat(checker.checkReadiness(null).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.emptySet()).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(ImmutableSet.of("test-ok", "test-unhealthy")).getStatus())
+        .isEqualTo(SERVICE_UNAVAILABLE.getStatusCode());
+    assertThat(checker.checkReadiness(Collections.singleton("test-ok")).getStatus())
+        .isEqualTo(OK.getStatusCode());
```

Comment:
Nit: seems like a candidate for splitting into two tests

Reply:
Converted to `@ParameterizedTest` in 41cc8a73883bbd8583bf41d19b6762769bc17ebf